If terminal is too small, output doesn't update properly

When running `nanobox deploy` output can spam smaller terminals. Likely not a big deal at all.

Exaggerated example (happens with terminals 80 columns wide as well(common default width)):
![image](https://cloud.githubusercontent.com/assets/2653109/21993176/bef81732-dbd6-11e6-94fd-e5b7df6ae9cd.png)

Probably more related to this than a new issue with compiling, but when trying to debug why `nanobox compile-app` isn't compiling the app, nanobox panics.

To view the panic, terminal has to be <40 characters wide.

```
panic: runtime error: slice bounds out of range

goroutine 51 [running]:
panic(0x936ca0, 0xc420014150)
	/home/travis/.gimme/versions/go1.7.3.linux.amd64/src/runtime/panic.go:500 +0x1a1
github.com/nanobox-io/nanobox/util/display.(*Summarizer).handleLog(0xc420096300, 0xc420318c40, 0x35)
	/home/travis/gopath/src/github.com/nanobox-io/nanobox/util/display/summary.go:225 +0x1de
github.com/nanobox-io/nanobox/util/display.(*Summarizer).run(0xc420096300)
	/home/travis/gopath/src/github.com/nanobox-io/nanobox/util/display/summary.go:146 +0x18f
created by github.com/nanobox-io/nanobox/util/display.(*Summarizer).Start
	/home/travis/gopath/src/github.com/nanobox-io/nanobox/util/display/summary.go:68 +0x3f
```

Ubuntu 16.10
Nanobox version 0.9.0 ()

--- a/util/display/summary.go
+++ b/util/display/summary.go
@@ -225,11 +225,6 @@ func (s *Summarizer) handleLog(data string) {
 			continue
 		}
 
-		availableLen := s.windowWidth - (len(s.Label) + len(s.Prefix) + 5)
-		if s.windowWidth > 0 && len(line) > availableLen {
-			line = line[:availableLen] + "..."
-		}
-
 		s.detail = line
 		s.reset()
 		s.print()
@@ -315,7 +310,19 @@ func (s *Summarizer) print() {
 
 
 	header := fmt.Sprintf("%s%s %s :\n", s.Prefix, TaskSpinner[s.spinIdx], s.Label)
+
+	// truncate the header 
+	availableLen := s.windowWidth - 5
+	if s.windowWidth > 0 && len(header) > availableLen {
+		header = header[:availableLen] + "...\n"
+	}
+
 	detail := fmt.Sprintf("%s  %s\n", s.Prefix, s.detail)
+
+	// truncate the details
+	if s.windowWidth > 0 && len(detail) > availableLen {
+		detail = detail[:availableLen] + "..."
+	}
 
 	// todo: add progress estimator
 

